### PR TITLE
[SPARK-5457][SQL] Add missing DSL for ApproxCountDistinct.

### PR DESCRIPTION
**Pull request by ueshin**

### CODE REVIEW

**Comment by marmbrus** [comment]
Sorry, @rxin just reorganized the DSL as part of our dataframe update.  Mind merging?

File: sql/core/src/main/scala/org/apache/spark/sql/Dsl.scala [hunk 1]
```diff
@@ -105,6 +105,8 @@ object Dsl {
   def countDistinct(expr: Column, exprs: Column*): Column =
     CountDistinct((expr +: exprs).map(_.expr))
 
+  def approxCountDistinct(e: Column, rsd: Double = 0.05): Column =
```
Comment (rxin) [inline sql/core/src/main/scala/org/apache/spark/sql/Dsl.scala]: we need java support. can you remove the default value and just have two methods?

Comment (ueshin) [inline sql/core/src/main/scala/org/apache/spark/sql/Dsl.scala]: Ah, I see. I'll fix it.

**Comment by rxin** [comment]
Thanks. Merging in master.

**Pull request closed without merging**